cmd/kubeadm/app/phases/kubelet/flags.go

```diff
@@ -61,9 +61,31 @@ func WriteKubeletDynamicEnvFile(nodeRegOpts *kubeadmapi.NodeRegistrationOptions,
 	}
 	stringMap := buildKubeletArgMap(flagOpts)
 	argList := kubeadmutil.BuildArgumentListFromMap(stringMap, nodeRegOpts.KubeletExtraArgs)
-	envFileContent := fmt.Sprintf("%s=%s\n", constants.KubeletEnvFileVariableName, strings.Join(argList, " "))
 
-	return writeKubeletFlagBytesToDisk([]byte(envFileContent), kubeletDir)
+	kubeConfigArgs := map[string]string{
+		"config": filepath.Join(kubeletDir, constants.KubeletConfigurationFileName),
+	}
+	kubeConfigArgsList := kubeadmutil.BuildArgumentListFromMap(kubeConfigArgs, map[string]string{})
+
+	kubeconfigArgs := map[string]string{
+		"bootstrap-kubeconfig": filepath.Join(kubeConfigDir, constants.KubeletBootstrapKubeConfigFileName),
+		"kubeconfig":           filepath.Join(kubeConfigDir, constants.KubeletKubeConfigFileName),
+	}
+	kubeconfigArgsList := kubeadmutil.BuildArgumentListFromMap(kubeconfigArgs, map[string]string{})
+
+	kubeadmFlags := strings.Join(argList, " ")
+	kubeconfigFlags := strings.Join(kubeConfigArgsList, " ")
+	configFlags := strings.Join(kubeconfigArgsList, " ")
+	efc := envFileContent(kubeadmFlags, kubeconfigFlags, configFlags)
+	return writeKubeletFlagBytesToDisk([]byte(efc), kubeletDir)
+}
+
+func envFileContent(kubeadmFlags, kubeconfigFlags, configFlag string) string {
```

Comment:
probably best to unroll this function into the caller?

Reply:
sure, i have no preference on this style